### PR TITLE
feat: Mirror Node Contract Queries

### LINKED ISSUE #1167: Add a query to estimate the gas of a contract call
### Description
Hedera's EVM equivalence enables developers to deploy and execute smart contracts using standard EVM-compatible tools. Additionally, the Hedera SDK allows the execution of smart contract transactions (state-changing and query-based) via protobuf messages submitted to Consensus Nodes.

However, unlike typical EVM flows where transactions are encoded with RLP and submitted to a JSON-RPC endpoint, Hedera’s SDK can introduce friction for developers transitioning from other ecosystems.

### Problem

Today, the SDK does not support a way to determine the gas of a contract call. 

EVM developers, accustomed to estimating gas requirements as part of their workflow, face challenges using the SDK. Specifically:

- The `ContractCall` transaction type allows setting a gas limit but does not support gas estimation.
- This absence of gas estimation is a core gap, making the SDK less developer-friendly for those familiar with Ethereum-like environments.

### Solution

Add a query that estimates the gas for a contract call. 

1. Add a new query type
2. Add `getGasCost` Support to `ContractCall`

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by 0xivanov**
**Description**:

A new `MirrorNodeContractQuery` class has been introduced to support EVM gas estimation and simulation. This query class provides two distinct types of queries:

1. **`MirrorNodeContractCallQuery`** – Executes a contract call and returns the result, simulating the execution of the contract.
2. **`MirrorNodeContractEstimateGasQuery`** – Estimates the gas usage for a contract call, allowing developers to understand the cost of executing a transaction.

The inputs for these queries match those of `ContractExecuteTransaction`, making it easy for developers to calculate the cost of a follow-up transaction.

The `MirrorNodeContractQuery` issues an HTTP request to the `api/v1/contracts/call` endpoint of the Mirror Node Web3 module, which supports both gas estimation and transient simulation of state-changing transactions. This enables accurate cost predictions and helps developers optimize their contract interactions before executing them on the network.

**Related issue(s)**:

Fixes #1167

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: contract_id.go [hunk 1]
```diff
@@ -212,6 +212,47 @@ func (id ContractID) _ToProtobuf() *services.ContractID {
...
+	if index == -1 {
+		return nil, errors.New("invalid mirrorUrl format")
+	}
+	mirrorUrl = mirrorUrl[:index]
+
+	var url string
+	protocol := httpsString
+	port := ""
+
+	if client.GetLedgerID().String() == "" {
```
Comment (0xivanov) [inline contract_id.go:231]: @ivaylonikolov7 fyi

File: mirror_node_contract_estimate_gas_query.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+// This specifies the price of each unit of gas used in the transaction.
+func (query *MirrorNodeContractEstimateGasQuery) SetGasPrice(gasPrice int64) *MirrorNodeContractEstimateGasQuery {
+	query.setGasPrice(gasPrice)
+	return query
+}
+
+// Returns gas estimation for the EVM execution
+func (query *MirrorNodeContractEstimateGasQuery) Execute(client *Client) (uint64, error) {
+	return query.estimateGas(client)
+}
```
Comment (gsstoykov) [inline mirror_node_contract_estimate_gas_query.go:74]: Is there any difference between `MirrorNodeContractEstimateGasQuery` and `MirrorNodeContractCallQuery` if not for this `execute` method. You are housing the logic for `estimateGas` and the `call` in the parent which is `query`. Going down this path you would have to extend the logic in `query` while creating files and structs which just hold all the properties of the `query` and just adding an unnecessary layer of setters and getters. I would see the need of having `MirrorNodeContractEstimateGasQuery` and `MirrorNodeContractCallQuery` if they held at least some specific fields or functionalities.

**Review by gsstoykov — COMMENTED** [review]
Few nits and needs a little design tweak. Otherwise LG.

File: mirror_node_contract_call_query.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
+package hiero
+
+// SPDX-License-Identifier: Apache-2.0
+
+// MirrorNodeContractCallQuery returns a result from EVM transient simulation of read-write operations.
+type MirrorNodeContractCallQuery struct {
```
[the remark this answers is not in the document]
Comment (0xivanov) [reply mirror_node_contract_call_query.go:6]: The idea is that one query is for estimation and the other for call. Anyhow, we already agreed on the names with nana, simi and richard.

File: mirror_node_contract_estimate_gas_query.go (hunk 2 above)
Comment (0xivanov) [reply mirror_node_contract_estimate_gas_query.go:74]: We are doing it solely for the naming. I initially had them be 1 query, which had `estimate` and `call` methods, but Simi and Nana wanted the query name to say what the query does - hence how we have 2 queries - `EstimateGasQuery` and `CallQuery`.

File: account_id.go [hunk 4]
```diff
@@ -324,6 +324,9 @@ const (
 	EvmAddress
 )
 
+var httpString string = "http"
+var httpsString string = "https"
```
[the remark this answers is not in the document]
Comment (0xivanov) [reply account_id.go:328]: I might as well remove them, as they are not needed anymore i think

File: mirror_node_contract_query.go [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+	// The gas limit
+	gasLimit *int64
+	// The gas price
+	gasPrice *int64
+	// The block number for the simulation
+	blockNumber *int64
+}
+
+// setContractID sets the contract instance to call
+func (query *mirrorNodeContractQuery) setContractID(contractID ContractID) {
```
Comment (gsstoykov) [inline mirror_node_contract_query.go:40]: nit: Would also be good to differentiate between  the `query` naming here as we also have `query` in the transaction flow just for ease for code navigation and etc.

Comment (0xivanov) [reply mirror_node_contract_query.go]: named `mirrorNodeContractQuery` for clarity

File: account_id.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (0xivanov) [reply account_id.go]: removed

**Review by gsstoykov — APPROVED** [review]
LGTM.

**Pull request merged**